x-pack/plugins/ml/common/util/job_utils.js

```diff
@@ -105,18 +105,20 @@ export function isModelPlotChartableForDetector(job, detectorIndex) {
     const dtr = dtrs[detectorIndex];
     const functionName = dtr.function;
 
-    // Model plot can be charted for any of the functions which map to ES aggregations,
+    // Model plot can be charted for any of the functions which map to ES aggregations
+    // (except rare, for which no model plot results are generated),
     // plus varp and info_content functions.
     isModelPlotChartable =
-      mlFunctionToESAggregation(functionName) !== null ||
-      [
-        'varp',
-        'high_varp',
-        'low_varp',
-        'info_content',
-        'high_info_content',
-        'low_info_content',
-      ].includes(functionName) === true;
+      functionName !== 'rare' &&
+      (mlFunctionToESAggregation(functionName) !== null ||
+        [
+          'varp',
+          'high_varp',
+          'low_varp',
+          'info_content',
+          'high_info_content',
+          'low_info_content',
+        ].includes(functionName) === true);
```

Comment:
this could be converted to use the constants from `kibana/x-pack/plugins/ml/common/constants/aggregation_types.ts`

Reply:
Agreed, although will save this for a follow-up as I wanted to keep the code churn in this fix to a minimum. I suspect there are a lot of places in the code where we use hard-coded function names which could be switched to the constants from `kibana/x-pack/plugins/ml/common/constants/aggregation_types.ts`.